rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs

```diff
@@ -58,26 +87,26 @@ async fn invoke_pre_config(
 
 async fn invoke_post_config(
     cfg: &mut ConfigBag,
-    context: InterceptorContext,
+    mut before_serialization: InterceptorContext<BeforeSerialization>,
     interceptors: Interceptors,
 ) -> Result<Output, SdkError<Error, HttpResponse>> {
-    let context = Phase::construction(context)
-        // Before serialization
-        .include(|ctx| interceptors.read_before_serialization(ctx, cfg))?
-        .include_mut(|ctx| interceptors.modify_before_serialization(ctx, cfg))?
-        // Serialization
-        .include_mut(|ctx| {
-            let request_serializer = cfg.request_serializer();
-            let request = request_serializer
-                .serialize_input(ctx.take_input().expect("input set at this point"))?;
-            ctx.set_request(request);
-            Result::<(), BoxError>::Ok(())
-        })?
-        // After serialization
-        .include(|ctx| interceptors.read_after_serialization(ctx, cfg))?
-        // Before retry loop
-        .include_mut(|ctx| interceptors.modify_before_retry_loop(ctx, cfg))?
-        .finish();
+    // Before serialization
+    handle_err!(before_serialization => interceptors.read_before_serialization(&before_serialization, cfg));
+    handle_err!(before_serialization => interceptors.modify_before_serialization(&mut before_serialization, cfg));
```

Comment:
If you use the incorrect invocation of the `handle_err!` macro _(with brackets vs. without)_, does it produce an error? Or, do you have to be careful?

Reply:
It will fail to compile.